### PR TITLE
Add parallel scale test, and cleanups

### PR DESCRIPTION
**Pull request by grahamwhaley**
Add a test to measure scalability of parallel container launch/delete.
To integrate nicely with the existing code base, make a number of cleanups as well.

### CODE REVIEW

**Comment by obedmr** [comment]
/me testing locally and then also in Azure CI

**Comment by obedmr** [comment]
@grahamwhaley thanks for all this, it's working really good, I have a some questions or comments: 

if I already run the `metrics/scaling/k8s_scale.sh` and I want to generate the report, is it possible? I tried running the `makereport.sh` from [report](https://github.com/clearlinux/cloud-native-setup/tree/master/metrics/report) dir and it's not generating it, it says that it's not finding the required files.

I found that the `grabdata.sh` script is running the both `k8s_scale.sh` and `k8s_parallel.sh`, shouldnt this be only for grabing pre-generated data?

File: metrics/scaling/k8s_parallel.sh [hunk 1]
```diff
@@ -0,0 +1,293 @@
+#!/bin/bash
```
Comment (dklyle) [inline metrics/scaling/k8s_parallel.sh]: I think we should either combined this with k8s_scale.sh or extract more common tooling to share, one of the scripts is going to get stale quickly. But we can start this way.

File: metrics/lib/common.bash [hunk 2]
```diff
@@ -42,93 +44,6 @@ check_cmds()
 	done
 }
 
-# This function performs a docker pull on the image names
```
Comment (dklyle) [inline metrics/lib/common.bash]: happy to see all this go!

**Comment by grahamwhaley** [comment]
np @obedmr 
- `grabdata.sh` maybe should be called more correctly `generatedata.sh`. It actually runs the tests to get a new set of data.
- `makereport.sh` looks in subdirs of `results` for data to process - if you don't hand-move the data from `grabdata.sh` into (well named) subdirs, then it will not find anything to process.... the subdirs is because `makereport.sh` can (and mostly is used to) handle multiple sets of data - so you can compare changes etc.

So, fairly normal operation is like:
- grabdata
- mkdir results/run1; mv *.json run1
- grabdata
- mkdir results/run2; mv *.json run2
- makereport.sh

iyswim.

@dklyle did suggest `grabdata` could make a datestamped dir and put the results in it automatically - which, I think we could do pretty easily. I can do that in another PR (and add some better docs) if we want.

File: metrics/scaling/k8s_parallel.sh (hunk 1 above)
Comment (grahamwhaley) [reply metrics/scaling/k8s_parallel.sh]: I agree. My gut says share the common code via libs, depending on just how similar tests are. A balancing act.
Yeah, I pulled out the api code in this PR, and I expect we will pull out and factor our/share more code as we add more tests.

File: metrics/scaling/k8s_parallel.sh [hunk 3]
```diff
@@ -0,0 +1,293 @@
...
+# sleep and timeout times for k8s actions, in seconds
+wait_time=${wait_time:-30}
+delete_wait_time=${delete_wait_time:-600}
+use_api=${use_api:-yes}
+
+# Set some default metrics env vars
+TEST_ARGS="runtime=${RUNTIME}"
+TEST_NAME="k8s parallel"
+
+get_num_nodes() {
```
[the remark this answers is not in the document]
Comment (grahamwhaley) [reply metrics/scaling/k8s_parallel.sh]: good spot - I forgot to dedup them both! Done.

**Comment by dklyle** [comment]
I think you pushed the wrong branch, the last commit if for your other PR

**Comment by grahamwhaley** [comment]
Oops, had that extra commit in for some local testing, and forgot/didn't spot it was still there! apologies!
pushed.. hope I didn't mess it up..

**Pull request merged**